include/unifex/stop_immediately.hpp

```diff
@@ -413,6 +417,14 @@ struct _stream<SourceStream, Values...>::type {
     operation<Receiver> connect(Receiver&& receiver) && {
       return operation<Receiver>{stream_, (Receiver &&) receiver};
     }
+    template<typename Receiver>
+    operation<Receiver> connect(Receiver&& receiver) & {
+      return operation<Receiver>{stream_, (Receiver &&) receiver};
+    }
+    template<typename Receiver>
+    operation<Receiver> connect(Receiver&& receiver) const& {
+      return operation<Receiver>{stream_, (Receiver &&) receiver};
+    }
```

Comment:
I'm pretty sure it doesn't make sense to allow repeatedly running the cleanup operation here.
This cleanup operation implementation shared references to the stream object and assumes that it will only be run once - the state-machine does not allow for running the cleanup operations multiple times.

In general, I think cleanup operations should generally not be repeatable. They will often have side-effects that release resources and should only be run once.

It also seems suspicious that each of these are performing the same

Reply:
Destructors should only be run once and should be run on a const object.

I do not think that const connect maps directly to repeatable.

Reply:
> Destructors should only be run once and should be run on a const object.

Are you equating the `cleanup()` operation to a destructor here?

I guess for ordinary types, you can call the destructor on a const object, however the object ceases to be a const object inside the destructor (I still want to be able to mutate data members inside a destructor). Also, a destructor consumes an object.

So I can perhaps see an argument for allowing the `cleanup()` method to be declared const-rvalue. However, there isn't an easy way to declare that the method was const-rvalue callable but that inside the method the data-members were all mutable without sprinkling const_cast everywhere.

I think the qualification on the `cleanup()` method is a separate concern from the qualification of the `connect()` method on the sender returned from `cleanup()`.

I don't want people to think that it's safe to call `connect()` multiple times on this sender. The current way we indicate that it is safe to call `connect()` multiple times is by adding a const-lvalue overload of `connect()`.